substrate/frame/revive/src/vm/pvm/env.rs

```diff
@@ -719,6 +719,11 @@ pub mod env {
 	) -> Result<(), TrapReason> {
 		self.charge_gas(RuntimeCosts::DepositEvent { num_topic, len: data_len })?;
 
+		self.emitted_events += 1;
+		if self.emitted_events > limits::NUM_EMITTED_EVENTS {
```

Comment:
I've added a limit of maximum 512 events. The amount of bytes they contain should be limited below to `self.ext.max_value_size()`. Let me know if we should add extra limits 🙏